src/services/myMap.js

```diff
@@ -0,0 +1,140 @@
+const { promisify } = require('util');
+
+const { sale: SALE } = require('../config');
+
+// eslint-disable-next-line no-extend-native
+Array.prototype.myMap = async (oldArray, callback) => {
+  const newArray = [];
+
+  for (let i = 0; i < oldArray.length; i++) {
+    // eslint-disable-next-line no-await-in-loop
+    newArray.push(await callback(oldArray[i], i, oldArray));
+  }
+
+  return newArray;
+};
+
+function randomNumber(min, max) {
+  return Math.floor(Math.random() * (max - min + 1)) + min;
+}
+
+async function callbackFunction(callback) {
+  const promise = promisify(setTimeout);
+
+  await promise(SALE.TIME_GENERATE_SALE);
+
+  const number = randomNumber(SALE.MIN, SALE.MAX);
+
+  if (number >= SALE.NOT_MORE_THAN) {
+    return callback('Incorrect sale!');
+  }
+
+  return callback(null, number);
+}
+
+function promiseFunction() {
+  return new Promise((resolve, reject) => {
+    setTimeout(() => {
+      const result = randomNumber(SALE.MIN, SALE.MAX);
+
+      if (result >= SALE.NOT_MORE_THAN) {
+        return reject(new Error('Incorrect sale: SALE'));
+      }
+      return resolve(result);
+    }, SALE.TIME_GENERATE_SALE);
+  });
+}
+
+async function methodCallback() {
+  const result = await callbackFunction(async (err, res) => {
+    if (err) {
+      const func = await methodCallback();
+      return func;
+    }
+
+    return res;
+  });
+
+  return result;
+}
+
+async function methodPromise() {
+  let result;
+
+  try {
+    result = await promiseFunction();
+  } catch (err) {
+    result = await methodPromise();
+  }
+
+  return result;
+}
+
+async function methodAsync() {
+  let result;
+
+  try {
+    result = await promiseFunction();
+  } catch {
+    result = await methodAsync();
+  }
+
+  return result;
+}
```

Comment:
Can be replaced by shorter variant
```
try {
  return await promiseFunction();
} catch (err) {
  return methodPromise();
}
```
Also notice that there is no `await` in the `catch` statement